### PR TITLE
Tighten tempfile() return types

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
A first stab. We also have to split `_TextMode` and `_BinaryMode`, similar to how they are split in `builtins.pyi`. We probably don't need to handle the read case, as a read-only temp file makes no sense. This could even help catch further bugs.

File: stdlib/tempfile.pyi [hunk 1]
```diff
@@ -30,13 +30,14 @@ TMP_MAX: int
 tempdir: str | None
 template: str
 
-_StrMode: TypeAlias = Literal["r", "w", "a", "x", "r+", "w+", "a+", "x+", "rt", "wt", "at", "xt", "r+t", "w+t", "a+t", "x+t"]
-_BytesMode: TypeAlias = Literal["rb", "wb", "ab", "xb", "r+b", "w+b", "a+b", "x+b"]
+_TextMode: TypeAlias = Literal["r", "w", "a", "x", "r+", "w+", "a+", "x+", "rt", "wt", "at", "xt", "r+t", "w+t", "a+t", "x+t"]
```
Comment (AlexWaygood) [inline stdlib/tempfile.pyi:33]: Does `_typeshed.OpenTextModeUpdating` not work here?

Comment (srittau) [reply stdlib/tempfile.pyi:33]: Probably, I want to look into reusing the literals in `builtins.pyi`.

**Comment by srittau** [comment]
Depends on #10233.

**Comment by srittau** [comment]
This is the last prerequisite for trying #10229 again.

**Pull request merged**